mafiasi/registration/views.py

```diff
@@ -58,63 +84,54 @@ def request_account(request):
                         'domain': domain,
                         'info_form': info_form,
                     })
+        else:
+            return TemplateResponse(request, 'registration/request_account.html', {
+                'form_principal': form if is_primary else PrimaryRegisterForm(),
+                'form_other': OtherRegisterForm() if is_primary else form,
+                'is_primary': is_primary,
+            })
 
-    else:
-        form = RegisterForm()
+    form = PrimaryRegisterForm(initial={'domain': settings.PRIMARY_DOMAIN})
 
     return TemplateResponse(request, 'registration/request_account.html', {
-            'form': form,
+            'form_principal': form,
+            'form_other': OtherRegisterForm(),
+            'is_primary': True,
         })
 
 
 @require_POST
 def additional_info(request):
-    if request.POST:
-        form = AdditionalInfoForm(request.POST)
-        if form.is_valid():
-            return _finish_account_request(request, {
-                        'action': 'request_account',
-                        'account': form.cleaned_data['account'],
-                        'domain': form.cleaned_data['domain'],
-                        'first_name': form.cleaned_data['first_name'],
-                        'last_name': form.cleaned_data['last_name'],
-                        'yeargroup_pk': form.cleaned_data['yeargroup'].pk
-                    })
-
-    if not 'domain' in form.cleaned_data or not 'account' in form.cleaned_data:
-        return redirect('registration_request_account')
-    else:
-        domain = form.cleaned_data['domain']
+    form = AdditionalInfoForm(request.POST)
+    if form.is_valid():
         account = form.cleaned_data['account']
+        domain = form.cleaned_data['domain']
+        return _finish_account_request(request, {
+                    'action': 'request_account',
+                    'domain': domain,
+                    'first_name': form.cleaned_data['first_name'],
+                    'last_name': form.cleaned_data['last_name'],
+                    'yeargroup_pk': form.cleaned_data['yeargroup'].pk,
+                    'email': f'{account}@{domain}',
+                    'account': account,
+                })
 
-    if domain == settings.PRIMARY_DOMAIN:
-        template = 'registration/require_info.html'
-    else:
-        template = 'registration/require_info_other.html'
-    form.prefill(account, domain)
-
-    return TemplateResponse(request, template, {
-        'account': account,
-        'domain': domain,
-        'info_form': form,
-    })
 
+def request_successful(request):
+    return TemplateResponse(request, 'registration/request_successful.html')
```

Comment:
The message in `request_successful.html` (if the account exists …) does not make sense for email-based registrations.